Utils.js

```diff
@@ -0,0 +1,159 @@
+
+function addSkill () {
+  /* Adds skill to storage and to current table.
+   */
+  var skillName = $('#skill_name').val();
+  $('#skill_name').val('');
+  if(skillName === ""){
+    return;
+  }
+  // If not for this if, hundreds of empty skills would be created with Enter key.
+  // To fix.
+  var skillsDict = {};
+  skillsDict[skillName] = 0;
+  allSkills.push(skillName);
+  chrome.storage.sync.set({skillsArrayId: allSkills});
+  chrome.storage.sync.set(skillsDict);
+  debugAddingSkill(skillName);
+
+  newSkillToTable(allSkills.length - 1);
+}
+
+function clearSkills () {
+  /* Erases all skills.
+   */
+  var emptyArray = [];
+  chrome.storage.sync.set({skillsArrayId: emptyArray});
+}
+
+function debugSkillsExp(index) {
+  /* Writes to consol all skills and it's exp.
+   */
+  var skillName = allSkills[index];
+  chrome.storage.sync.get([skillName], function (result) {
+    extension_log(skillName + ": " + result[skillName]);
+  });
+  if (index + 1 < allSkills.length) {
+    debugSkillsExp(index + 1);
+  }
+}
+
+function removeSkill(skillNr) {
+    /* Removes skill from allSkills table
+     *    * and stores modified table in Chrome Storage.
+     *       */
+    allSkills.splice(skillNr, 1);
+    chrome.storage.sync.set({skillsArrayId: allSkills});
+    resetHTMLTable();
+}
+
+function exportStorage () {
+  let keyList = allSkills.slice();
+  keyList.push("skillsArrayId");
+  extension_log("KEYLIST:");
+  extension_log(keyList);
+  chrome.storage.sync.get(keyList, function (result) {
+    extension_log("Hi");
+    $('#storage_stringified').html(JSON.stringify(result));
+  });
+}
+
+function importStorage () {
+  var text = $('#storage_stringified').val();
+  var rightBracketPos = 0;
+  while(text[rightBracketPos] != "]" && rightBracketPos < text.length){
+    rightBracketPos++;
+  }
+  if(rightBracketPos < text.length && text[rightBracketPos] == "]"){
+    var toSkills = text.slice(0, rightBracketPos + 1);
+    var toExp = text.slice(rightBracketPos + 1, text.length);
+    var newSkills = JSON.parse(toSkills);
+    var newExp = JSON.parse(toExp);
+    chrome.storage.sync.set({skillsArrayId: newSkills}, function () {
+      allSkills = newSkills;
+      extension_log(allSkills.length);
+
+      function setExp (i) {
+        if(i < allSkills.length){
+          var skillName = allSkills[i];
+          var skillExp = newExp[i];
+          var skillsDict = {};
+          skillsDict[skillName] = skillExp;
+          chrome.storage.sync.set(skillsDict, setExp(i+ 1));
+        }
+        else{
+          setTimeout(function() {
+            resetHTMLTable();
+          },100);
+        }
+      }
+      setExp(0);
+    });
+  }
+  else{
+    extension_log("Incorrect JSONs.");
+  }
+}
+
+function fillExpTable() {
+    /* Fills expTable with numbers according to a certain formula.
+     *    */
+  console.log("Dupa");
+  for (var i = 0; i < maxLevel; i++) {
+          expTable[i] = Math.floor(i * i / 2);
+        }
+  console.log(expTable[3]);
```

Comment:
Tego typu też - Console.logi które można zostawić to takie które mówią o tym, że jakaś funkcja lub moduł jest poprawnie zainicjowany.